src/Analyser/NodeScopeResolver.php

```diff
@@ -2292,6 +2293,25 @@ static function (?Type $offsetType, Type $valueType, bool $optional) use (&$arra
 
 			$this->callNodeCallbackWithExpression($nodeCallback, new BooleanOrNode($expr, $leftResult->getFalseyScope()), $scope, $context);
 
+			return new ExpressionResult(
```

Comment:
i'm not confident that i've done the right thing here, just winged it based on what the BooleanOrConstantConditionRule is doing and the node structure it relies on.

i would not be surprised if i have overlooked something in this code